### PR TITLE
gst: add get request pad function

### PR DESCRIPTION
**Pull request by brucekim**
@tinyzimmer Getting request pad is necessary for multiplexer like compositor, aggregator to get a template pad that is created on demand, not automatic.

I found other one also wants to this function in go-gst. 
https://github.com/tinyzimmer/go-gst/issues/14

Following is excerpt from gstreamer manual regarding request pad.
https://gstreamer.freedesktop.org/documentation/application-development/basics/pads.html?gi-language=c

> Request pads
An element can also have request pads. These pads are not created automatically but are only created on demand. This is very useful for multiplexers, aggregators and tee elements. Aggregators are elements that merge the content of several input streams together into one output stream. Tee elements are the reverse: they are elements that have one input stream and copy this stream to each of their output pads, which are created on request. Whenever an application needs another copy of the stream, it can simply request a new output pad from the tee element.
> The gst_element_get_request_pad () method can be used to get a pad from the element based on the name of the pad template.

### CODE REVIEW

File: gst/gst_element.go [hunk 1]
```diff
@@ -501,3 +501,21 @@ func (e *Element) URIHandler() URIHandler {
 func (e *Element) RemovePad(pad *Pad) bool {
 	return gobool(C.gst_element_remove_pad(e.Instance(), pad.Instance()))
 }
+
+// GetRequestPad gets a request pad from the element based on the name of the pad template.
+// Unlike static pad, request pads are not automatically but are only created on demand.
```
Comment (tinyzimmer) [inline gst/gst_element.go:506]: "are not automatically **created,** but instead, only on demand"

Or something like that, sounds better

Comment (brucekim) [reply gst/gst_element.go:506]: Thanks for your comment. I supplement description.
How about following?

// GetRequestPad gets a request pad from the element based on the name of the pad template.
// Unlike static **pads**, request pads **are not created automatically but are only created on demand
// For example, audiomixer has sink template, 'sink_%u', which is used for creating multiple sink pads on demand so that it performs mixing of audio streams by linking multiple upstream elements on it's sink pads created on demand.**

**Comment by brucekim** [comment]
@tinyzimmer Is it ok for you about latest revision, supplement comment.

**Comment by tinyzimmer** [comment]
I still have the question about that `pad= nil` - What's it for?

**Comment by brucekim** [comment]
`pad = nil ` in ReleaseRequestPad() is to let go handle it by garbage collecting. ReleaseRequestPad() is intentionally to make request pad free.

For example, let consider a scenario in which four 'testvideosource' elements are lined with a compositor. Each testvideosource would be liked with the compositor through a sink pad respectively. The sink pad is type of request pad. 

Then, let consider that , we want to remove one testvideosource linked. In order to do that, first of all, we unlink the testvideosource. In addition to that, we need to release resource of the sink pad that was connected to the testvideosource.

Here, I think RequestRequestPad() is used to free the sink pad. Along with it, `pad=nil` make the reference of the pad at go side null.

In summary,
* Removing a element linked to a request pad requires release of the resource related to the element including request pad.
* For that one, request pad should also be released as well by using ReleaseRequestPad() I think.

It would be appreciated if you describe in detail regarding your concern.

**Comment by tinyzimmer** [comment]
My concern is that it feels very opinionated and forces something that would otherwise happen naturally, depending on the structure of the code. Perhaps if you show me an example of you using it this way it would give me better context. It might make more sense to me what you are describing. 

---

From the gstreamer docs you linked:

> This does not unref the pad. If the pad was created by using gst_element_request_pad, gst_element_release_request_pad needs to be followed by gst_object_unref to free the pad.

That's fine. Once the pad leaves scope it will eventually get unrefed by the finalizer placed on it from `FromGstPadUnsafeFull`. In my head the only thing that would keep that from happening is the _way_ in which you write the code. If the structure of someone's code demands that `pad = nil` after Releasing, otherwise finalizers won't run, then I feel it would be better for the caller themselves do that instead of the function. 

So basically, it feels possible the user's code can be refactored in a way to make this not necessary. If I see an example that proves otherwise that might sway me.

**Comment by brucekim** [comment]
Thanks for your review.
Yes, I agree with your concern. I checked the finalizer eventually does unref, which was placed from `FromGstPadUnsafeFull()`.
I am somehow confusing whether you think GetRequestPad() is not necessary for go-gst.
I updated an example and a patch that removes `pad=nil` line.

It would be appreciated if you share how you think about it.

**Comment by tinyzimmer** [comment]
LGTM - I'm a bit busy today but will merge later
